src/security/Handshake.cc

```diff
@@ -550,23 +551,26 @@ Security::HandshakeParser::ParseCertificate(const SBuf &raw, Security::CertPoint
     pCert = nullptr;
     debugs(83, 2, "TLS parsing is not supported without OpenSSL. " << raw);
 #endif
+    return pCert;
 }
 
 void
 Security::HandshakeParser::parseServerCertificates(const SBuf &raw)
 {
+#if USE_OPENSSL
     Parser::BinaryTokenizer tkList(raw);
     const SBuf clist = tkList.pstring24("CertificateList");
     Must(tkList.atEnd()); // no leftovers after all certificates
 
     Parser::BinaryTokenizer tkItems(clist);
     while (!tkItems.atEnd()) {
-        Security::CertPointer cert;
-        ParseCertificate(tkItems.pstring24("Certificate"), cert);
-        serverCertificates.push_back(cert);
+        if (Security::CertPointer cert = ParseCertificate(tkItems.pstring24("Certificate")))
+            serverCertificates.push_back(cert);
         debugs(83, 7, "parsed " << serverCertificates.size() << " certificates so far");
     }
-
+#else
+    debugs(83, 2, "TLS certificates parsing is not supported without OpenSSL");
```

Comment:
This may generate a warning about unused `raw`. Let's add the length of that buffer to debugs().

Also, I do not think it is useful to shout about lack of OpenSSL support every time we parse a server Hello message. Let's raise the debug level.

Finally, we do not need to explain exactly why we cannot parse the certificates -- the code speaks for itself.

I suggest the following sketch to address all of the above:

```
debugs(83, 7, "no support for CertificateList parsing; ignoring " << raw.len() << " bytes");
```